File: sorting/golang/select.go
package main

import (
	"fmt"
)

func select_sort(s []int) []int {
	n := len(s)
	for i := 0; i < n; i++ {
		min := i
		for j := i + 1; j < n; j++ {
			if s[min] > s[j] {
				min = j
			}
		}
		s[i], s[min] = s[min], s[i]
	}
	return s
}

func main() {
	s := []int{1, 3, 9, 0, 2, 4, 7}
	fmt.Println(s, "---default")
	fmt.Println(select_sort(s), "---select")
}

File: sorting/python/select.py
#!/usr/local/bin/python2
# -*- coding: UTF-8 -*-

def select_sort(arr):
    n = len(arr)
    for i in range(0, n):
        min = i
        for j in range(i + 1, n):
            if arr[min] > arr[j]:
                min = j
        arr[min], arr[i] = arr[i], arr[min]
    return arr

if __name__ == "__main__":
    arr = [1, 3, 9, 0, 2, 4, 7]

    print arr, "---default"
    print select_sort(arr), "---select"
